numpyro/contrib/module.py

```diff
@@ -433,3 +435,132 @@ def random_haiku_module(
         _update_params(params, new_params, prior)
     nn_new = partial(nn.func, new_params, *nn.args[1:], **nn.keywords)
     return nn_new
+
+
+def nnx_module(name, nn_module):
+    """
+    Declare a :mod:`~flax.nnx` style neural network inside a
+    model so that its parameters are registered for optimization via
+    :func:`~numpyro.primitives.param` statements.
+
+    Given a flax NNX ``nn_module``, to evaluate the module, we directly call it.
+    In a NumPyro model, the pattern will be::
+
+        # Eager initialization outside the model
+        module = nn_module(...)
+
+        # Inside the model
+        net = nnx_module("net", module)
+        y = net(x)
+
+    :param str name: name of the module to be registered.
+    :param flax.nnx.Module nn_module: a pre-initialized `flax nnx` Module instance.
+    :return: a callable that takes an array as an input and returns
+        the neural network transformed output array.
+    """
+    try:
+        from flax import nnx
+    except ImportError as e:
+        raise ImportError(
+            "Looking like you want to use flax.nnx to declare "
+            "nn modules. This is an experimental feature. "
+            "You need to install the latest version of `flax` to use this feature. "
+            "It can be installed with `pip install git+https://github.com/google/flax.git`."
+        ) from e
+
+    graph_def, eager_params_state, eager_other_state = nnx.split(
+        nn_module, nnx.Param, nnx.Not(nnx.Param)
+    )
+
+    eager_params_state_dict = nnx.to_pure_dict(eager_params_state)
+
+    module_params = None
+    if eager_params_state:
+        module_params = numpyro.param(name + "$params")
+    if module_params is None:
+        module_params = numpyro.param(name + "$params", eager_params_state_dict)
+
+    eager_other_state_dict = nnx.to_pure_dict(eager_other_state)
+
+    mutable_holder = None
+    if eager_other_state_dict:
+        mutable_holder = numpyro_mutable(name + "$state")
+    if mutable_holder is None:
+        numpyro_mutable(name + "$state", {"state": eager_other_state_dict})
```

Comment:
mutable_holder = numpyro_mutable(...)

Reply:
fixed in https://github.com/pyro-ppl/numpyro/pull/1990/commits/18fe2a280a14b51556b4c054d7a53a6354a89d02